Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -303,19 +318,24 @@ void Launcher::for_each_handler_for_path(String const& path, Function<bool(Handl
 
     auto extension = LexicalPath::extension(path).to_lowercase();
     auto mime_type = mime_type_for_file(path);
+
     if (mime_type.has_value()) {
-        for_each_handler(mime_type.value(), m_mime_handlers, [&](auto const& handler) -> bool {
-            if (handler.handler_type != Handler::Type::Default || handler.mime_types.contains(mime_type.value()))
-                return f(handler);
-            return false;
-        });
+        if (!for_each_handler(mime_type.value(), m_mime_handlers, [&](auto const& handler) -> bool {
+                if (handler.handler_type != Handler::Type::Default || handler.mime_types.contains(mime_type.value()))
+                    return f(handler);
+                return false;
+            }))
+            goto handle_extension;
     } else {
-        for_each_handler(extension, m_file_handlers, [&](auto const& handler) -> bool {
-            if (handler.handler_type != Handler::Type::Default || handler.file_types.contains(extension))
-                return f(handler);
-            return false;
-        });
+        goto handle_extension;
```

Comment:
I think this is entirely unnecessary; please avoid `goto` at almost all costs.

Reply:
You're rightttt.